google/resource_container_cluster_test.go

```diff
@@ -12,6 +12,80 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
+// TestContainerClusterNodeVersionCustomizeDiffFuncNilReturn is doing unit test for cases where
+// the return have to be "nil".
+func TestContainerClusterNodeVersionCustomizeDiffFuncNilReturn(t *testing.T) {
+
+	cases := map[string]struct {
+		nameBeforePolicy            string
+		nodeVersionAfterPolicy      string
+		minMasterVersionAfterPolicy string
+	}{
+		"same version on create": {
+			nameBeforePolicy:            "",
+			nodeVersionAfterPolicy:      "version1",
+			minMasterVersionAfterPolicy: "version1",
+		},
+		"same version not on create": {
+			nameBeforePolicy:            "randomName",
+			nodeVersionAfterPolicy:      "version1",
+			minMasterVersionAfterPolicy: "version1",
+		},
+		"different version not on create": {
+			nameBeforePolicy:            "randomName",
+			nodeVersionAfterPolicy:      "version1",
+			minMasterVersionAfterPolicy: "otherVersion",
+		},
+	}
+
+	for tn, tc := range cases {
```

Comment:
If you use `t.Run` for each case creating a [subtest](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks), I believe that cases will get reported more cleanly.